src/axl.c

```diff
@@ -526,6 +533,16 @@ int AXL_Dispatch (int id)
         char* dest_dir = dirname(dest_path);
         mode_t mode_dir = axl_getmode(1, 1, 1);
         axl_mkdir(dest_dir, mode_dir);
+
+        /*
+         * Special case: The BB API checks if the destination path exists at
+         * its equivalent of AXL_Add() time.  That's why we do its "AXL_Add"
+         * here, after the full path to the file has been created.
+         */
+        if (xtype == AXL_XFER_ASYNC_BBAPI) {
+            char *src = kvtree_elem_key(elem);
+            axl_async_add_bbapi(id, src, dest);
```

Comment:
should we check the return code here? or does the axl_async_add_bbapi throw up an error if one is encountered?

Reply:
It will print an error via a failed `bb_check()`, but won't stop anything.  I just added a check in my latest push that will catch the error though,